### DISCUSSION TITLE
rustdesk binary executable was in wrong location (SysV init)

### DISCUSSION THREAD

**Discussion by rizalmart** [rustdesk/rustdesk#6011] [Feature Request]
### Bug Description

On latest rustdesk release the binary executable was in /usr/lib/rustdesk instead of /usr/bin. But the desktop "Exec=" and "TryExec=" parameter was set to just rustdesk which means the executable should be in /usr/bin

### How to Reproduce

Just double-click rustdesk desktop files

### Expected Behavior

It shows no such file or directory error message

### Operating system(s) on local side and remote side

Debian Linux

### RustDesk Version(s) on local side and remote side

1.2.3

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/49640119/48780e9b-8af5-4c8b-9d9b-6fd7a3d21949)

**Comment by dan9er** [comment]
This has been an issue since 1.2.0, for both the x86_64 and aarch64 debs.

How was this glaring issue overlooked? Why has it, when someone finally reported it months later, been quickly kicked to discussion without a word?

EDIT: I thought twice about my tone, and I'm sorry. I felt since there's a company behind this and you're charging for pro licences, I expected a higher standard. It may just be a one-person company.

**Comment by rustdesk** [comment]
@sahilyeole please help to debug with them why below does not work on their system.

https://github.com/rustdesk/rustdesk/blob/c6ace470e3b42fb83a1dce913e29c9ac76a242f3/res/DEBIAN/postinst#L10

**Comment by beelchester** [comment]
For me, it is there at both `usr/bin` and `usr/lib/rustdesk` on debian.

@rizalmart @dan9er 
Can you please run this script (as sudo) and tell the output?

```bash
#!/bin/bash

INITSYS=$(ls -al /proc/1/exe | awk -F' ' '{print $NF}' | awk -F'/' '{print $NF}')
	
if [ "systemd" == "$INITSYS" ]; then
	ln -s /usr/lib/rustdesk/rustdesk /usr/bin/rustdesk
else
	echo "error: $INITSYS"
fi
```

**Comment by dan9er** [comment]
Yeah, I see the problem now. We're not using `systemd` (Before you ask, here's [some](https://nosystemd.org/) [reasons](https://suckless.org/sucks/systemd) [why](https://ewontfix.com/14/)), so that symlink line is never run. I don't know what rizalmart is running, but I'm running [Devuan](https://www.devuan.org/) which uses SysV `init`.

That said, I don't see why systemd is mandatory to do a simple symlink. That line (and the related ones inside the other files) should be moved outside the if block.

I can see the system service won't work on my system, but is it required to do basic connections? Can it run as a user service?

**Reply by beelchester** [reply]
>That said, I don't see why systemd is mandatory to do a simple symlink. That line (and the related ones inside the other files) should be moved outside the if block.

Yes, I also think symlink can be moved out if block, and this issue will be solved.

>I can see the system service won't work on my system, but is it required to do basic connections? Can it run as a user service?

You should be able to connect without system service but some features like unattended remote access will not work.

**Comment by beelchester** [comment]
@rustdesk Should we move the `ln` line above if condition? Can it cause any other problem?

Currently, for non systemd users symlink is not generating because of this condition. Hence causing this issue.
https://github.com/rustdesk/rustdesk/blob/c6ace470e3b42fb83a1dce913e29c9ac76a242f3/res/DEBIAN/postinst#L7-L10

**Reply by rustdesk** [reply]
It is ok to move above `if`. Though no service installed.

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/commit/84a21d72b9bda27e070ecd6daede80699475c53a
